operator/pkg/controlplane/controlplane.go

```diff
@@ -0,0 +1,133 @@
+package controlplane
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+	clientset "k8s.io/client-go/kubernetes"
+	clientsetscheme "k8s.io/client-go/kubernetes/scheme"
+
+	operatorv1alpha1 "github.com/karmada-io/karmada/operator/pkg/apis/operator/v1alpha1"
+	"github.com/karmada-io/karmada/operator/pkg/util"
+	"github.com/karmada-io/karmada/operator/pkg/util/apiclient"
+)
+
+// PatchManifest defines a function to patch deployment
+type PatchManifest func(cfg *operatorv1alpha1.KarmadaComponents, deployment *appsv1.Deployment)
```

Comment:
What's `PatchManifest` used for?

Reply:
It is used to patch manifest after parsing resource yaml.